frontend/javascripts/dashboard/dataset/dataset_settings_view.tsx

```diff
@@ -339,12 +339,18 @@ class DatasetSettingsView extends React.PureComponent<PropsWithFormAndRouter, St
       this.state.activeDataSourceEditMode === "simple" ? "advanced" : "simple",
     );
 
-    const afterForceUpdateCallback = () =>
+    const afterForceUpdateCallback = () => {
       // Trigger validation manually, because fields may have been updated
-      form
-        .validateFields()
-        .then((formValues) => this.submit(formValues))
-        .catch((errorInfo) => this.handleValidationFailed(errorInfo));
+      // and defer the validation as it is done asynchronously by antd or so.
+      setTimeout(
+        () =>
+          form
+            .validateFields()
+            .then((formValues) => this.submit(formValues))
+            .catch((errorInfo) => this.handleValidationFailed(errorInfo)),
+        0,
+      );
+    };
```

Comment:
I managed to fix https://github.com/scalableminds/webknossos/issues/8172. The reason for the error is (as correctly described in the issue) first: The syncValidator in `dataset_settings_data_tab.tsx` needs to use the up-to-date value of the datasetFields and the form field validation failed due to an `outOfDate` error signaled by antd. The reason for this seems to be, that the validation was done on outdated values. 

After some trial and error I came up with the fix of simply deferring the whole validation as the debugger showed that the form validation was done by antd in some kind of asynchronous context or so. Therefore, by forcefully making the validation and submitting also asynchronous, it should be reliable executed after the form validation leading to up to date validated value in the `validateFields` call avoiding the bug described in #8172.

I hope this makes kinda sense. I did not dive deeper into why this resolved the bug, but it seems to work reliably, and I do not want to put more time into it.

Reply:
I found [this SO post](https://stackoverflow.com/questions/62731874/form-validatefields-fails-inside-onvalueschange-in-ant-design-forms) referencing [this antd issue](https://github.com/ant-design/ant-design/issues/26747) which are not about this exact use case but pretty similar. Since a maintainer suggests to use `setTimeout` it's probably fine as a workaround for now.